src/de/value.rs

```diff
@@ -0,0 +1,150 @@
+
+    fn expecting(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "A RON value")
+    }
+
+    fn visit_bool<E>(self, v: bool) -> Result<Self::Value, E>
+        where E: Error
+    {
+        Ok(Value::Bool(v))
+    }
+
+    fn visit_i64<E>(self, v: i64) -> Result<Self::Value, E>
+        where E: Error,
+    {
+        self.visit_f64(v as f64)
+    }
+
+    fn visit_u64<E>(self, v: u64) -> Result<Self::Value, E>
+        where E: Error
+    {
+        self.visit_f64(v as f64)
+    }
+
+    fn visit_f64<E>(self, v: f64) -> Result<Self::Value, E>
+        where E: Error
+    {
+        Ok(Value::Number(( /*TODO*/ )))
+    }
+
+    fn visit_char<E>(self, v: char) -> Result<Self::Value, E>
+        where E: Error
+    {
+        Ok(Value::Char(v))
+    }
+
+    fn visit_str<E>(self, v: &str) -> Result<Self::Value, E>
+        where E: Error
+    {
+        self.visit_string(v.to_owned())
+    }
+
+    fn visit_string<E>(self, v: String) -> Result<Self::Value, E>
+        where E: Error
+    {
+        Ok(Value::String(v))
+    }
+
+    fn visit_bytes<E>(self, v: &[u8]) -> Result<Self::Value, E>
+        where E: Error
+    {
+        self.visit_byte_buf(v.to_vec())
+    }
+
+    fn visit_byte_buf<E>(self, v: Vec<u8>) -> Result<Self::Value, E>
+        where E: Error
+    {
+        self.visit_string(String::from_utf8(v)
+            .map_err(|e| Error::custom(format!("{}", e)))?)
+    }
+
+    fn visit_none<E>(self) -> Result<Self::Value, E>
+        where E: Error
+    {
+        Ok(Value::Option(None))
+    }
+
+    fn visit_some<D>(self, deserializer: D) -> Result<Self::Value, D::Error>
+        where D: Deserializer<'de>,
+    {
+        Ok(Value::Option(Some(Box::new(deserializer.deserialize_any(ValueVisitor)?))))
+    }
+
+    fn visit_unit<E>(self) -> Result<Self::Value, E>
+        where E: Error
+    {
+        Ok(Value::Unit)
+    }
+
+    fn visit_newtype_struct<D>(self, deserializer: D) -> Result<Self::Value, D::Error>
+        where D: Deserializer<'de>
+    {
+        deserializer.deserialize_any(ValueVisitor)
+    }
+
+    fn visit_seq<A>(self, mut seq: A) -> Result<Self::Value, A::Error>
+        where A: SeqAccess<'de>
+    {
+        let mut vec = Vec::new();
+        if let Some(cap) = seq.size_hint() {
+            vec.reserve_exact(cap);
+        }
+
+        while let Some(x) = seq.next_element()? {
+            vec.push(x);
+        }
+
+        Ok(Value::Seq(vec))
+    }
+
+    fn visit_map<A>(self, mut map: A) -> Result<Self::Value, A::Error>
+        where A: MapAccess<'de>
+    {
+        let mut res: BTreeMap<Value, Value> = BTreeMap::new();
+
+        while let Some(entry) = map.next_entry()? {
+            res.insert(entry.0, entry.1);
+        }
+
+        Ok(Value::Map(res))
+    }
+
+    fn visit_enum<A>(self, data: A) -> Result<Self::Value, A::Error>
+        where A: EnumAccess<'de>
+    {
+        let (value, variant) = data.variant()?;
+
+        Ok(Value::Enum(variant, Box::new(value)))
```

Comment:
@dtolnay I couldn't figure out how to do deal with enums here yet.

Reply:
Never mind, there is no way I could even detect an enum, so I removed `Value::Enum`.